File: .examples/.js
(() => {
  class Person {
    constructor(name, age) {
      this.name = name;
      this.age = age;
    }

    greet() {
      return `Hello, my name is ${this.name} and I am ${this.age} years old.`;
    }
  }

  const calculateYearOfBirth = (currentYear, age) => currentYear - age;

  let currentYear = new Date().getFullYear();
  let personName = 'Alice';
  let personAge = 30;

  const person = new Person(personName, personAge);
  const yearOfBirth = calculateYearOfBirth(currentYear, personAge);

  person.greet();
  console.log(`${person.name} was born in ${yearOfBirth}.`);
})();

File: .examples/.py
class Person:
  def __init__(self, name: str, age: int):
    self.name = name
    self.age = age

  def greet(self) -> str:
    return f"Hello, my name is {self.name} and I am {self.age} years old."

def calculate_year_of_birth(current_year: int, age: int) -> int:
  return current_year - age

current_year = 2023
person_name = "Alice"
person_age = 30

person = Person(person_name, person_age)
year_of_birth = calculate_year_of_birth(current_year, person_age)

print(person.greet())
print(f"{person.name} was born in {year_of_birth}.")
